server/functions/src/refreshcasestats.ts

```diff
@@ -0,0 +1,371 @@
+import axios from "axios";
+import * as admin from "firebase-admin";
+import * as functions from "firebase-functions";
+import { SERVING_REGION } from "./config";
+import * as who from "./who";
+
+// THis seems wrong to initializeApp both here and index.ts?
+//admin.initializeApp();
+//const db = admin.firestore();
+// The following line causes the Firebase emulator to say:
+//  "It appears your code is written in Typescript, which must be compiled before emulation."
+//const db = admin.firestore();
+// This crazy mess makes it work.
+
+var db: admin.firestore.Firestore;
+export function setDb(newDb: admin.firestore.Firestore) {
+  db = newDb;
+}
+
+// Triggers to reject case stats update for unexpectedly large increase
```

Comment:
Nit: "trigger" in the context of a Cloud Function is a thing that causes the Cloud Function to run. I'd prefer to not overload the name.

I'd describe this type of parameter as a "safety threshold" or (less common, but I like this term) a "fuse".

Reply:
Many of the comments in this file are copy/pasted from refreshcasestats.java.! Made some wording changes.